frontend/test/pytest/test_gradient.py

```diff
@@ -1780,5 +1780,35 @@ def fn(x):
         assert np.allclose(res_pattern_partial, expected)
 
 
+@pytest.mark.parametrize("argnums", [0, 1, (0, 1)])
+@pytest.mark.parametrize("transform_qjit", [False, True])
+def test_grad_argnums(argnums, transform_qjit):
+    """Tests https://github.com/PennyLaneAI/catalyst/issues/1477"""
+    @qml.qnode(device=qml.device("lightning.qubit", wires=4), interface="jax")
+    def circuit(inputs, weights):
+        qml.AngleEmbedding(features=inputs, wires=range(4), rotation="X")
+        for i in range(1, 4):
+            qml.CRX(weights[i - 1], wires=[i, 0])
+        return qml.expval(qml.PauliZ(wires=0))
+
+    if transform_qjit:
+        circuit = qjit(circuit)
+
+    weights = jnp.array([3.0326467, 0.98860157, 1.9887222])
+    inputs = jnp.array([0.9653214, 0.31468165, 0.63302994])
+
+    def compare_structure_and_value(o1, o2):
+        return tree_structure(o1) == tree_structure(o2) and \
+            tree_all(tree_map(jnp.allclose, o1, o2))
+
+    result = grad(circuit, argnums=argnums)(weights, inputs)
+    expected = jax.grad(circuit, argnums=argnums)(weights, inputs)
+    assert compare_structure_and_value(result, expected)
+
+    _, result = value_and_grad(circuit, argnums=argnums)(weights, inputs)
+    _, expected = jax.value_and_grad(circuit, argnums=argnums)(weights, inputs)
```

Comment:
Ideally we might want to compare against a computation not involving Catalyst at all, which can be achieved like this (applying `jax.grad` function is a sort of Catalyst/JAX hybrid):
```suggestion
    expected = jax.grad(circuit.original_function, argnums=argnums)(weights, inputs)
    assert compare_structure_and_value(result, expected)

    _, result = value_and_grad(circuit, argnums=argnums)(weights, inputs)
    _, expected = jax.value_and_grad(circuit.original_function, argnums=argnums)(weights, inputs)
```